account_payment_term_partner_holiday/models/account_invoice.py

```diff
@@ -16,7 +16,11 @@ def _onchange_payment_term_date_invoice(self):
 
     @api.onchange("date_due")
     def _onchange_date_due_account_payment_term_partner_holiday(self):
-        if self.date_due and self.partner_id:
+        """Is necessary to re-calculate only when payment_term_id is not set
+        to prevent date_due is changed again and apply +10 days for example
+        according to payment term.
+        """
```

Comment:
If you are adding a docstring, it would be better to explain the full behavior of the method. I think this would make it clearer:

```suggestion
        """Recompute the due date to the next available date according to
        the holiday periods set in the partner.
        
        It must only be re-calculated when a payment term is not set.
        This prevents the due date to be changed again and that another 
        given number of days are added according to what is set on the 
        payment term.
        """
```

Or something like that, if you think it is not completely right...